src/overlays/actors/ovl_En_Fd/z_en_fd.c

```diff
@@ -21,61 +41,796 @@ const ActorInit En_Fd_InitVars = {
+    Player* player = PLAYER;
+    ColliderBody* body;
+
+    if (this->collider.base.acFlags & 2 || EnFd_CheckHammer(this, globalCtx)) {
+        this->collider.base.acFlags &= ~2;
+        if (this->invincibilityTimer != 0) {
+            return false;
+        }
+        body = &this->collider.list[0].body;
+        if (body->acHitItem != NULL && (body->acHitItem->toucher.flags & 0x80)) {
+            return false;
+        }
+
+        if (!EnFd_SpawnCore(this, globalCtx)) {
+            return false;
+        }
+        this->invincibilityTimer = 30;
+        this->actor.flags &= ~1;
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_FLAME_DAMAGE);
+        func_80032C7C(globalCtx, &this->actor);
+        return true;
+    } else if (DECR(this->attackTimer) == 0 && this->collider.base.atFlags & 2) {
+        this->collider.base.atFlags &= ~2;
+        if (this->invincibilityTimer != 0) {
+            return false;
+        }
+
+        if (this->collider.base.atFlags & 4) {
+            return false;
+        }
+        this->attackTimer = 30;
+        Audio_PlayActorSound2(&player->actor, NA_SE_PL_BODY_HIT);
+        func_8002F71C(globalCtx, &this->actor, this->actor.speedXZ + 2.0f, this->actor.yawTowardsLink, 6.0f);
+    }
+    return false;
+}
+
+/**
+ * Determines if `actor` is within an acceptable range for `this` to be able to "see" `actor`
+ * `actor` must be within 400 units of `this`, `actor` must be within +/- 40 degrees facing angle
+ * towards `actor`, and there must not be a collision poly between `this` and `actor`
+ */
+s32 EnFd_CanSeeActor(EnFd* this, Actor* actor, GlobalContext* globalCtx) {
+    CollisionPoly** colPoly;
+    u32 bgId;
+    Vec3f colPoint;
+    s16 angle;
+    s32 pad;
+
+    // Check to see if `actor` is within 400 units of `this`
+    if (Math_Vec3f_DistXYZ(&this->actor.posRot.pos, &actor->posRot.pos) > 400.0f) {
+        return false;
+    }
+
+    // Check to see if the angle between this facing angle and `actor` is withing ~40 degrees
+    angle = (f32)Math_Vec3f_Yaw(&this->actor.posRot.pos, &actor->posRot.pos) - this->actor.shape.rot.y;
+    if (ABS(angle) >= 0x1C71) {
+        return false;
+    }
+
+    // check to see if the line between `this` and `actor` does not intersect a collision poly
+    if (func_8003DE84(&globalCtx->colCtx, &this->actor.posRot.pos, &actor->posRot.pos, &colPoint, &colPoly, 1, 0, 0, 1,
+                      &bgId)) {
+        return false;
+    }
+
+    return true;
+}
+
+Actor* EnFd_FindBomb(EnFd* this, GlobalContext* globalCtx) {
+    Actor* actor = globalCtx->actorCtx.actorList[ACTORTYPE_EXPLOSIVES].first;
+
+    while (actor != NULL) {
+        if (actor->params != 0 || actor->parent != NULL) {
+            actor = actor->next;
+            continue;
+        }
+
+        if (actor->id != ACTOR_EN_BOM) {
+            actor = actor->next;
+            continue;
+        }
+
+        if (EnFd_CanSeeActor(this, actor, globalCtx) != 1) {
```

Comment:
```suggestion
        if (EnFd_CanSeeActor(this, actor, globalCtx) != true) {
```
Kind of weird, but it's treated as a bool everywhere else.

Reply:
The only reason I'd be against this is that it's actually checking for a 1 versus when I usually use things as a bool it's a check against not zero or a check against zero.